akka-http-core/src/main/scala/akka/http/impl/engine/http2/framing/FrameRenderer.scala

```diff
@@ -91,7 +91,8 @@ private[http2] object FrameRenderer {
               b.putInt32(value)
 
               renderNext(remaining)
-            case Nil =>
+            case Nil   =>
+            case other => throw new IllegalArgumentException(s"Unexpected remaining: $other") // compiler completeness check pleaser
```

Comment:
Even though this shouldn't happen, could this result in that we toString on unknown stuff and possibly include sensitive (or too much) in logs? Similar for other places where the compiler completeness was added.

Reply:
Good thinking, I mostly thought that it would be good with details in case the should never happen actually happens, but better safe, at least on parse/render side of things. I'll take another pass through those and see.

Reply:
Done in 8d14ad6f4